app/set/directives/savedsearches/mode-context.directive.js

```diff
@@ -0,0 +1,113 @@
+module.exports = SavedSearchModeContext;
+
+SavedSearchModeContext.$inject = [];
+
+function SavedSearchModeContext () {
+    return {
+        restrict: 'E',
+        scope: {
+            savedSearch: '=',
+            filters: '='
+        },
+        controller: SavedSearchModeContextController,
+        templateUrl: 'templates/sets/savedsearches/mode-context.html'
+     };
+ }
+
+SavedSearchModeContextController.$inject = [
+    '$scope',
+    '$translate',
+    'NotificationEndpoint',
+    'SavedSearchEndpoint',
+    'Notify',
+    '_',
+    'ModalService'
+];
+function SavedSearchModeContextController(
+    $scope,
+    $translate,
+    NotificationEndpoint,
+    SavedSearchEndpoint,
+    Notify,
+    _,
+    ModalService
+) {
+    // Show Add Notification link
+    $scope.showNotificationLink = true;
+    $scope.canEdit = false;
+    $scope.notification = false;
+
+    $scope.saveNotification = saveNotification;
+    $scope.removeNotification = removeNotification;
+    $scope.editSavedSearch = editSavedSearch;
+    $scope.deleteSavedSearch = deleteSavedSearch;
+
+    activate();
+
+    function activate() {
+        $scope.canEdit = canEdit($scope.savedSearch);
+
+        NotificationEndpoint.query({set: $scope.savedSearch.id, ignore403: true, user: 'me'}, function (notifications) {
+            // show link if subscription does not exist
+            $scope.showNotificationLink = notifications.length === 0;
+            if (notifications.length) {
+                $scope.notification = notifications[0];
+            }
+        });
+    }
+
+    // Check if we can edit
+    function canEdit(savedSearch) {
+        return _.contains(savedSearch.allowed_privileges, 'update');
+    }
+
+    function editSavedSearch() {
+        ModalService.openTemplate('<saved-search-editor saved-search="savedSearch"></saved-search-editor>', 'set.edit_search_settings', 'star', $scope, false, false);
```

Comment:
This now works to open a modal. I think for single directive tag templates like this, this is probably the preferred method (vs a template file). Anything else should go into a template file.